qualtrix/client.py

```diff
@@ -61,10 +85,77 @@ def result_export():
                 "browser": result["values"]["QID17_BROWSER"],
                 "version": result["values"]["QID17_VERSION"],
                 "os": result["values"]["QID17_OS"],
-                "resolution": result["values"]["QID17_RESOLUTION"]
+                "resolution": result["values"]["QID17_RESOLUTION"],
             }
             answers.append(answer)
         except KeyError:
             pass
-    
+
     return answers
+
+
+def delete_session(session_id: str):
+    """
+    POST /surveys/{surveyId}/sessions/{sessionId}
+    body {
+        "close": "true"
+    }
+    """
+    r_body = {"close": "true"}
+
+    url = settings.BASE_URL + f"/surveys/{settings.SURVEY_ID}/sessions/{session_id}"
+    r = requests.post(url, headers=auth_header, json=r_body)
+
+    if r.status_code == 200:
+        return r.json()
+
+
+def finalize_session(session_id: str, response_id: str):
+    """
+    Business logic for ending session, pulling response, and posting to gdrive
+    """
+    # if not delete_session(session_id):
+    #     raise error.QualtricsError("Not Found sessionId") 
+
+    # The session deletion api attempts to delete a session, must poll for a response
+    response = ""
+    for i in range(5):
+        response = get_response(response_id)
+        if response:
+            break
+        else:
+            print("Response still loading...")
+        time.sleep(2)
+    survey_answers = {"status": "", "response": {}}
+
+    print(response)
+    if not response or not response["meta"]["httpStatus"] == "200 - OK":
+        raise error.QualtricsError("Survey response not found")
+
+    labels = response["result"]["labels"]
+
+    result = response["result"]
+
+    # Assign survey response status
+    # Qualtrics API returns poorly documented boolean as string - unsure if it returns anything else
+    try:
+        survey_answers["status"] = (
+            "Complete" if eval(labels["finished"]) else "Incomplete"
+        )
+    except:
+        survey_answers["status"] = labels["finished"]
+
+    answer = {
```

Comment:
should have this logic only once.

Reply:
Using a different value from the API response will help this. I was using a value that appeared to be either 'True' or 'False', but did not guarantee their values. There is a safer option that is either 1 or 0.

Reply:
Sorry I was unclear. The code to pull data from surveyresult is duplicated.

Reply:
Gotcha, put the duplicated code in a helper function